src/repositories/gitlab_updater.rs

```diff
@@ -0,0 +1,323 @@
+use crate::error::Result;
+use crate::{db::Pool, Config};
+use chrono::{DateTime, Utc};
+use log::{info, warn};
+use postgres::Client;
+use reqwest::{
+    blocking::Client as HttpClient,
+    header::{HeaderMap, HeaderValue, ACCEPT, AUTHORIZATION, USER_AGENT},
+};
+use serde::Deserialize;
+use std::sync::Arc;
+
+use crate::repositories::{Updater, APP_USER_AGENT};
+
+const GRAPHQL_UPDATE: &str = "query($ids: [ID!]!) {
+    projects(ids: $ids) {
+        nodes {
+            id
+            nameWithNamespace
+            lastActivityAt
+            description
+            starCount
+            forksCount
+            openIssuesCount
+        }
+    }
+}";
+
+const GRAPHQL_SINGLE: &str = "query($fullPath: ID!) {
+    project(fullPath: $fullPath) {
+        id
+        nameWithNamespace
+        lastActivityAt
+        description
+        starCount
+        forksCount
+        openIssuesCount
+    }
+}";
+
+/// How many repositories to update in a single chunk.
+const UPDATE_CHUNK_SIZE: usize = 5;
+
+pub struct GitlabUpdater {
+    client: HttpClient,
+    pool: Pool,
+}
+
+impl Updater for GitlabUpdater {
+    /// Returns `Err` if the access token has invalid syntax (but *not* if it isn't authorized).
+    /// Returns `Ok(None)` if there is no access token.
+    fn new(config: Arc<Config>, pool: Pool) -> Result<Option<Self>> {
+        let mut headers = HeaderMap::new();
+        headers.insert(USER_AGENT, HeaderValue::from_static(APP_USER_AGENT));
+        headers.insert(ACCEPT, HeaderValue::from_static("application/json"));
+
+        if let Some(token) = &config.gitlab_accesstoken {
```

Comment:
Assuming `gitlab_accesstoken` is actually a gitlab.com access token, this is going to send it to all the other gitlab instances too.

Reply:
that one I did miss :) 

while that could be seen as security problem, it also plainly doesn't work :)  (using the gitlab token on other platforms)

Reply:
Depending on what access it has, it's a pretty big security problem. Afaik gitlab only has very broad permission models, so the only way this would really be "safe" would be if it was an access token generated for an account with no access and dedicated just for this purpose (which, tbf, it should be anyway since it doesn't need any private access).

Reply:
Actually I think we never answered that one: how should I handle all possible gitlab tokens?